source-code/core/app/src/createReactiveQuery.ts

```diff
@@ -0,0 +1,48 @@
+import type { Message, MessageQueryApi } from "@inlang/messages"
+import { ReactiveMap } from "@solid-primitives/map"
+import { createEffect } from "./solid.js"
+
+/**
+ * Creates a query API for messages.
+	@@ -9,41 +9,43 @@ import { createEffect } from "./solid.js"
+ */
+export function createReactiveQuery(messages: () => Array<Message>): MessageQueryApi {
+	//let index: ReactiveMap<string, Message>
+	const index = new ReactiveMap<string, Message>(messages().map((message) => [message.id, message]))
+
+	createEffect(() => {
+		// TODO: update index
+		// q: replace all entries?
+		// q: just add new entries?
+		// q: remove old entries?
+		// setIndex(new ReactiveMap(messages().map((message) => [message.id, message])))
+	})
+
+	return {
+		create: ({ data }) => {
+			index.set(data.id, data)
```

Comment:
what do we want to do here if the id already exists? Currently we override the entry

Reply:
Using `Result` for potentially failing queries like create, update?

Reply:
so `create` should not override?

Reply:
and `delete` should "fail" if the id does not exist?

Reply:
Or maybe just return `true` or `false`?

Reply:
@inlang/team

Reply:
I vote for `Result`

Reply:
Ah you mean because the CRUD ops are `void`. Good point! Let's not leak `Result` into the ecosystem. Using a boolean that indicates whether an operation was successful seems indeed better!

- not leaking `Result` into the ecosystem -> better DX because "what is a result" is avoided
- boolean are easier to handle

Reply:
Using a boolean is likely also good for `createVariant` and `updateVariantPattern`

Reply:
I did already implement it yesterday. I will add tests tomorrow.

Reply:
I wrote all tests for `query` but have nothing changed for variants yet.

> Using a boolean is likely also good for `createVariant` and `updateVariantPattern`

`updateVariantPattern` has multiple ways of failing. There returning `false` seems wrong.